packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageModule.kt

```diff
@@ -0,0 +1,34 @@
+package expo.modules.image
+
+import android.graphics.drawable.Drawable
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.engine.DiskCacheStrategy
+import com.bumptech.glide.load.model.GlideUrl
+import com.facebook.react.bridge.Promise
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReactContextBaseJavaModule
+import com.facebook.react.bridge.ReactMethod
+import java.lang.Exception
+
+class ExpoImageModule(val context: ReactApplicationContext) : ReactContextBaseJavaModule(context) {
+  override fun getName() = "ExpoImageModule"
+
+  @ReactMethod
+  fun prefetch(url: String, promise: Promise) {
+    val glideUrl = GlideUrl(url)
+    var result : Drawable? = null
+    try {
+      result = Glide.with(context)
+        .load(glideUrl)
+        .diskCacheStrategy(DiskCacheStrategy.DATA)
+        .submit()
+        .get()
+    } catch (e: Exception) {
+      promise.reject("E_PREFETCH_FAILURE", "Could not prefetch image: ${e.message}", e)
+    }
+    if (result == null) {
+      promise.reject("E_PREFETCH_FAILURE", "Could not prefetch image.")
```

Comment:
Do we know when this happens and what are the reasons `null` is returned? 🤔 
```suggestion
      promise.reject("ERR_IMAGE_PREFETCH_FAILURE", "Failed to prefetch the image: ${url}.")
```

Reply:
I thought if java is not null safety it might be okay to check that the value is not null, but maybe if we don't have a good reason to do this, I shouldn't? 🤔

Reply:
The `get()` method returns a Java object of type `Drawable!`. So IMO it's a good practice to null-check this.